testcode/lib/tinyshell.c

```diff
@@ -340,6 +354,23 @@ static void cmd_rename(void)
         printf("rename failed: %s\n", strerror(errno));
 }
 
+static void cmd_exec(void)
+{
+    int st;
+    unsigned char *progname, *arguments;
+
+    progname = strtok(args, " \t\n");
+    if (! progname) {
+        puts("usage: exec <progname> [arguments]");
+        return;
+    }
+    arguments = strtok(NULL, "");
+
+    /*printf("exec: %s %s\n", progname, arguments ? arguments : "");*/
+    st = exec(progname, arguments);
```

Comment:
Assigning the return value to `st` is pointless -- it always is `-1`.  You just should cast it to `void`.